app/services/polymorphic_term_repository.rb

```diff
@@ -0,0 +1,36 @@
+class PolymorphicTermRepository < Struct.new(:id)
+  class << self
+    alias_method :orig_new, :new
+    def new(*args)
+      orig_new(*args).build
+    end
+
+    def find(*args)
+      orig_new(*args).find
+    end
+  end
+
+  def build
+    repository.new(id)
+  end
+
+  def find
+    repository.find(id)
+  end
+
+  private
+
+  def repository
+    return vocabulary_repository unless id.include?("/")
```

Comment:
I don't like this logic here, it feels sort of hard-coded for how important the switch is.  Can we make something that tells us when an id indicates a vocabulary vs. a term?  Isn't TermUri sort of on that path already?